@here/harp-examples/src/geojson.ts

```diff
@@ -0,0 +1,261 @@
+ * to read GeoJSON from a file or from a textarea in the page. A default style is applied for all
+ * possible geometry types.
+ */
+export namespace GeoJsonExample {
+    const editorWidth = "550px";
+    const map = createBaseMap();
+
+    setUpFilePicker();
+    setUpEditor();
+
+    const featuresDataSource = new FeaturesDataSource();
+    map.addDataSource(featuresDataSource).then(() => {
+        featuresDataSource.setStyleSet(getStyleSet());
+    });
+
+    function setUpEditor() {
+        const editorInput = document.querySelector("#editor textarea") as HTMLTextAreaElement;
+        editorInput.placeholder = `{
+    type: "FeatureCollection",
+    features:[
+
+    ]
+}`;
+        const updateButton = document.querySelector("#editor button") as HTMLButtonElement;
+        updateButton.addEventListener("click", () => {
+            const geojson = JSON.parse(
+                (document.querySelector("#editor textarea") as HTMLTextAreaElement).value
+            );
+            featuresDataSource.setFromGeojson(geojson);
+            map.update();
+        });
+    }
+
+    function getStyleSet(): StyleSet {
+        return [
+            {
+                when: "$geometryType == 'polygon'",
+                technique: "fill",
+                renderOrder: 10000,
+                attr: {
+                    color: "#7cf",
+                    transparent: true,
+                    opacity: 0.8,
+                    lineWidth: 1,
+                    lineColor: "#003344"
+                }
+            },
+            {
+                when: "$geometryType == 'polygon'",
+                technique: "solid-line",
+                renderOrder: 10001,
+                attr: {
+                    color: "#8df",
+                    metricUnit: "Pixel",
+                    lineWidth: 5
+                }
+            },
+            {
+                when: "$geometryType == 'point'",
+                technique: "circles",
+                renderOrder: 10002,
+                attr: {
+                    size: 10,
+                    color: "5ad"
+                }
+            },
+            {
+                when: "$geometryType == 'line'",
+                technique: "solid-line",
+                renderOrder: 10000,
+                attr: {
+                    color: "#8df",
+                    metricUnit: "Pixel",
+                    lineWidth: 5
+                }
+            }
+        ];
+    }
+
+    function processGeoJsonFile(file: File) {
+        const reader = new FileReader();
+        reader.onload = () => {
+            (document.querySelector(
+                "#editor textarea"
+            ) as HTMLTextAreaElement).value = reader.result as string;
+            featuresDataSource.setFromGeojson(JSON.parse(reader.result as string));
+            map.update();
+        };
+        if (file.type === "application/geo+json" || file.type === "application/json") {
+            reader.readAsBinaryString(file);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsBinaryString

> At that time, the `result` attribute contains the *raw binary* data from the file.

Are you sure you want to use this one ?

Shouldn't you use this:
https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsText ???